src/ipc/ipc3/handler.c

```diff
@@ -254,7 +254,10 @@ static int ipc_stream_pcm_params(uint32_t stream)
 		 */
 		pipe_dbg(pcm_dev->cd->pipeline,
 			 "ipc: ignore PCM param change request on an active pipeline");
-		return 0;
+		/*To resolve the HS playback not happening when
+		 *speaker is played first in tdm use case.
+		 */
+		break;
```

Comment:
well if you want to mux two streams on the same link, it's quite common that the tdm masks are set ahead of time and you end-up transmitting zeroes if one of the two streams is not used. So yes you need a mux.

Reply:
but it looks there's already a mux/deux in the path. I guess the only change needed is to split BUF1.1 into 2 buffers feeding into th DAIs right?

Reply:
> but it looks there's already a mux/deux in the path. I guess the only change needed is to split BUF1.1 into 2 buffers feeding into th DAIs right?

Splitting the BUF1.1 in to 2 buffers feeding in to the DAI's is not required as there is only one physical link which is represented as two logical DAI's.  That's why TDM mode is enabled to support multiple channels of different streams on single physical line simultaneously.  Here MUX is taking 4 channel data and placing them as per the matrix to produce 4 channel data for simultaneous playback of multiple streams.  H/W params is sent because driver detects as two DAI's available one for speaker and one for headset based on the topology. Also two logical DAI's are required in the topology to register two cards and allow simultaneous playback on speaker and headset.